packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -135,26 +167,38 @@ export const addEdge = (
 	if (existingEdge) return;
 
 	// Check if type is compatible
-	const allowedTypes = targetInputPort.type.split('|');
+	const outputTypes = sourceOutputPort.type.split('|');
+	const allowedInputTypes = targetInputPort.type.split('|');
+	const intersectionTypes = _.intersection(outputTypes, allowedInputTypes);
+
+	// Not supported if there are more than oen matches
+	if (intersectionTypes.length > 1) {
+		console.error(`Ambiguous matching types [${outputTypes}] to [${allowedInputTypes}]`);
+		return;
+	}
+
+	// Not supported if there is a mismatch
 	if (
-		!allowedTypes.includes(sourceOutputPort.type) ||
+		intersectionTypes.length === 0 ||
 		(!targetInputPort.acceptMultiple && targetInputPort.status === WorkflowPortStatus.CONNECTED)
 	) {
 		return;
 	}
 
 	// Transfer data value/reference
-	if (targetInputPort.acceptMultiple && targetInputPort.value && sourceOutputPort.value) {
-		targetInputPort.label = `${sourceOutputPort.label},${targetInputPort.label}`;
-		targetInputPort.value = [...targetInputPort.value, ...sourceOutputPort.value];
+	targetInputPort.label = sourceOutputPort.label;
```

Comment:
The port labelling part will likely need a second pass, not exactly sure what we want to show if a port has multiple types.

Reply:
Maybe a simple ` or `.

Reply:
There's a `getPortLabel` function in this file that does this

Reply:
getPortLabel happens "after" we set the port label if I recall - this here happens at a lower level.